client/client.go

```diff
@@ -253,30 +245,36 @@ func (c httpMetricClient) retryWithBackoff(
 	for i := 0; i < c.maxRetries; i++ {
 
 		var uploadURL, hash string
-		uploadURL, hash, err = c.GetUploadURL(metricFile, metricSampleURL, agentVersion, UID)
+		uploadURL, hash, err = c.GetUploadURL(metricFile, metricSampleURL, agentVersion, UID, i)
 		if err != nil {
 			log.Debugf("Client proxy or deployment YAML may be misconfigured.  Please check your client settings.")
 			log.Errorf("error encountered while retrieving upload location: %v", err)
 			continue
 		}
 
+		var awsRequestID string
+		var responseDump []byte
+		var dumpErr error
 		resp, err = c.buildAndDoRequest(metricFile, uploadURL, agentVersion, UID, hash)
-
-		if c.verbose {
-			reponseDump, err := httputil.DumpResponse(resp, true)
-			if err != nil {
-				log.Errorln(err)
+		if resp != nil {
+			awsRequestID = resp.Header.Get("X-Amz-Request-Id")
+			responseDump, dumpErr = httputil.DumpResponse(resp, true)
+			if dumpErr != nil {
+				log.Errorln(dumpErr)
 				continue
 			}
-			log.Infoln(string(reponseDump))
 		}
 
 		if err != nil && strings.Contains(err.Error(), "Client.Timeout exceeded") {
 			time.Sleep(getSleepDuration(i))
+			log.Infof("Put S3 Retry %d: Failed to put data to S3, X-Amzn-Requestid: %s", i, awsRequestID)
+			log.Debugln(string(responseDump))
 			continue
 		}
 
 		if resp == nil {
+			log.Infof("Put S3 Retry %d: Failed to put data to S3, X-Amzn-Requestid: %s", i, awsRequestID)
+			log.Debugln(string(responseDump))
```

Comment:
Will these debug statements always be emitted, or is the default log level info?

Reply:
The default log level is INFO. 
These messages only display if they set the CLOUDABILITY_LOG_LEVEL env variable to "DEBUG"

Reply:
Given that these logs are only emitted in error cases I think we might as well just always log them.

Otherwise during the support process when we would need them someone will have to go configure the agent and wait for the error to reoccur.

Reply:
Updated so we print the Status Message (ex: 403 Forbidden) in INFO mode. 
Leaving the whole log dump to DEBUG to prevent sensitive information being dumped in logs in INFO mode

Reply:
The debug in this block will never contain anything since it's based off the data in resp.

Reply:
I also think the infof the statusMessage and the awsRequestID will also be nil here too. I can remove them, but in this case I am not sure what the error message should be? 
"Put S3 Retry %d: Failed to put data to S3. Nil Response". 
That log line does not give us anything super helpful...